### PR TITLE
Fix hidden TextInput on Android/Expo

### PR DESCRIPTION
**Pull request by davista123**
This pull request is a fix to issue #578, #435 and #680. 
Currently, on android the text input is hidden when the keyboard is focused. I provide a fix by leveraging the react-native-keyboard-spacer library. The fix renders only on android/expo as shown below:

``` javascript
renderKeyboardSpacer() {
   if (Platform.OS === 'android') {
      return (<KeyboardSpacer />);
    }
    return null; 
}
```

### CODE REVIEW

**Comment by xcarpentier** [comment]
@davista123 Thanks :)
Is that possible to remove this file `package-lock.json`?
We already using yarn.

File: src/GiftedChat.js [hunk 1]
```diff
@@ -573,6 +586,7 @@ GiftedChat.propTypes = {
   loadEarlier: PropTypes.bool,
   onLoadEarlier: PropTypes.func,
   isLoadingEarlier: PropTypes.bool,
+  renderKeyboardSpacer: PropTypes.func,
```
Comment (xcarpentier) [inline src/GiftedChat.js]: Why this props if you don't use it?

Comment (davista123) [reply src/GiftedChat.js]: I use it in the _render()_ function of GiftedChat.js

```javascript
render() {
  if (this.state.isInitialized === true) {
      return (
        <View style={styles.container}>
          <ActionSheet ref={(component) => (this._actionSheetRef = component)}>
            <View style={styles.container} onLayout={this.onMainViewLayout}>
              {this.renderMessages()}
              {this.renderInputToolbar()}
            </View>
          </ActionSheet>
         {this.renderKeyboardSpacer()}
        </View>
      );
    }
```

**Comment by jeaye** [comment]
Any update on this one? Having gifted chat do this by default would be great.

**Comment by MrGurns** [comment]
Subscribing.

**Comment by watadarkstar** [comment]
Any updates on this @davista123  @xcarpentier ? What still needs to be done?

**Comment by camin-mccluskey** [comment]
Any update on this? Currently this library works perfectly on iOS and suffers from this keyboard hiding issue on Android for me. I've tried everything to push the textInput up over the keyboard but nothing has worked.

**Comment by SathishSaminathan** [comment]
try this..
<View  style={{flex:1}}>
          <GiftedChat

            messages={this.state.messages} 
            onSend={messages => this.onSend(messages)}
            user={{
              _id: 1,
            }}
              ></GiftedChat>

            <KeyboardAvoidingView behavior={'padding'} keyboardVerticalOffset={80}/>
        </View>

**Comment by SathishSaminathan** [comment]
`<View style={{flex: 1}}>
        <GiftedChat
          messages={this.state.messages}
          onSend={messages => this.onSend(messages)}
          user={{
            _id: 1,
          }}></GiftedChat>

        <KeyboardAvoidingView
          behavior={'padding'}
          keyboardVerticalOffset={80}
        />
      </View>`

**Comment by FromSunNews** [comment]
> thử cái này xem sao..
> 
> <View style={{flex: 1}}> <GiftedChat messages={this.state.messages} onSend={messages => this.onSend(messages)} user={{ _id: 1, }}>
> 
> ```
>     <KeyboardAvoidingView
>       behavior={'padding'}
>       keyboardVerticalOffset={80}
>     />
>   `</View>`
> ```

I appreciate you, thiss work for me

**Pull request closed without merging**